pkg/kubelet/volume/nfs/nfs_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nfs
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+)
+
+func TestCanSupport(t *testing.T) {
+	plugMgr := volume.PluginMgr{}
+	plugMgr.InitPlugins(ProbeVolumePlugins(), volume.NewFakeHost("fake", nil, nil))
+	plug, err := plugMgr.FindPluginByName("kubernetes.io/nfs")
+	if err != nil {
+		t.Errorf("Can't find the plugin by name")
+	}
+	if plug.Name() != "kubernetes.io/nfs" {
+		t.Errorf("Wrong name: %s", plug.Name())
+	}
+	if !plug.CanSupport(&api.Volume{VolumeSource: api.VolumeSource{NFS: &api.NFSVolumeSource{}}}) {
+		t.Errorf("Expected true")
+	}
+	if plug.CanSupport(&api.Volume{VolumeSource: api.VolumeSource{}}) {
+		t.Errorf("Expected false")
+	}
+}
+
+type fakeNFSMounter struct {
+	FakeMounter mount.FakeMounter
+}
+
+func (fake *fakeNFSMounter) Mount(server string, source string, target string, readOnly bool) error {
+	flags := 0
+	if readOnly {
+		flags |= mount.FlagReadOnly
+	}
+	fake.FakeMounter.MountPoints = append(fake.FakeMounter.MountPoints, mount.MountPoint{Device: server, Path: target, Type: "nfs", Opts: nil, Freq: 0, Pass: 0})
+	return fake.FakeMounter.Mount(fmt.Sprintf("%s:%s", server, source), target, "nfs", 0, "")
+}
+
+func (fake *fakeNFSMounter) Unmount(target string) error {
+	fake.FakeMounter.MountPoints = []mount.MountPoint{}
+	return fake.FakeMounter.Unmount(target, 0)
+}
+
+func (fake *fakeNFSMounter) List() ([]mount.MountPoint, error) {
+	list, _ := fake.FakeMounter.List()
+	return list, nil
+}
+
+func (fake *fakeNFSMounter) IsMountPoint(dir string) (bool, error) {
```

Comment:
No, I filed a separate bug for it.  Not your problem.

On Thu, Mar 19, 2015 at 11:19 AM, Chris Alfonso <notifications@github.com>
wrote:

> In pkg/kubelet/volume/nfs/nfs_test.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4601#discussion_r26782845>
> :
>
> > +	}
> > +	fake.FakeMounter.MountPoints = append(fake.FakeMounter.MountPoints, mount.MountPoint{Device: server, Path: target, Type: "nfs", Opts: nil, Freq: 0, Pass: 0})
> > +	return fake.FakeMounter.Mount(fmt.Sprintf("%s:%s", server, source), target, "nfs", 0, "")
> > +}
> > +
> > +func (fake *fakeNFSMounter) Unmount(target string) error {
> > +	fake.FakeMounter.MountPoints = []mount.MountPoint{}
> > +	return fake.FakeMounter.Unmount(target, 0)
> > +}
> > +
> > +func (fake *fakeNFSMounter) List() ([]mount.MountPoint, error) {
> > +	list, _ := fake.FakeMounter.List()
> > +	return list, nil
> > +}
> > +
> > +func (fake *fakeNFSMounter) IsMountPoint(dir string) (bool, error) {
>
> As part of this PR?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4601/files#r26782845>
> .
>